libkbfs/favorites.go

```diff
@@ -120,15 +123,16 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 	// Fetch a new list if:
 	//  * The user asked us to refresh
 	//  * We haven't fetched it before
-	//  * The user wants the list of favorites.  TODO: use the cached list
-	//    once we have proper invalidation from the server.
-	if req.refresh || f.cache == nil || req.favs != nil {
+	//  * It's stale
+	if req.refresh || f.cache == nil || time.Now().After(f.cacheExpireTime) {
 		folders, err := kbpki.FavoriteList(req.ctx)
 		if err != nil {
 			return err
 		}
 
 		f.cache = make(map[Favorite]bool)
+		f.cacheExpireTime = libkb.ForceWallClock(time.Now()).Add(time.
```

Comment:
Please make the expire time interval into a global const in this file, for easier tweaking in the future.